cclib/method/ddec.py

```diff
@@ -271,27 +272,94 @@ def calculate(self, indices=None, fupdate=0.05):
             self.radial_grid_r.append(r)
 
         # Carry out step 3 of DDEC6 algorithm [Determine conditioned charge density and tau]
-        self.logger.info("Conditioning charge densities.")
+        self.logger.info("Conditioning charge densities. (Step 3/7)")
         self.condition_densities()
 
+        # STEP 4-7
+        # Optimize weights on each grid point
+        self._kappa = [0.0 for x in self.data.atomnos]
+        update_kappa = False
+        self.N_A = []
+        self.u_A = []
+        steps = 4
+        cycle = 0
+        while steps < 8:
```

Comment:
Think about writing this in an unrolled way with helper functions.

Reply:
I've unrolled steps 4 and 7 and left steps 5 and 6 in the while loop, because in these two steps, step index is not incremented if update_kappa is true. Thanks.